### PR TITLE
Generic Taskcluster task naming

### LINKED ISSUE #573: Ensure naming for runs is consistent
I noticed that the teacher training run is called `teacher` in some runs and  `teacher-1` in others. It will make it hard to compare them.

Please check all names for both offline and online Taskcluster runs and ensure they are consistent. We should also try to make them consistent with old GCP experiments but the teacher training recipe was different back then and it's ok that it's called differently (`teacher-all`, `teacher-finetuned`). Except that all other runs should have the same naming.

Examples:
recent fix: https://wandb.ai/teklia/565-lt-en
online: https://wandb.ai/moz-translations/en-ru/runs/rry2yzfq
taskcluster offline: https://wandb.ai/moz-translations/en-ca
old Snakemake experiment: https://wandb.ai/moz-translations/nl-en

### PR DESCRIPTION
**Pull request by vrigal**
Required by #519 (making `build_task_name` supports evaluation tasks labels tags from Taskcluster)

Closes #573

### CODE REVIEW

File: tracking/translations_parser/utils.py [hunk 1]
```diff
@@ -50,14 +50,16 @@
...
-    # Else try to parse an evaluation label with importer, dataset and auugmentation
-    match = EVAL_REGEX.match(tag)
+    if match is None:
+        # Else try to parse an evaluation label with importer, dataset and auugmentation
+        match = EVAL_REGEX.match(tag)
     if not match:
         raise ValueError(tag)
     groups = match.groupdict()
-    return groups["model"], groups["importer"], groups["dataset"], groups["aug"]
+    # Always keep the index in the model name (teacher-1/2 → teacher-1)
```
Comment (eu9ene) [inline tracking/translations_parser/utils.py:59]: FYI we're going to rename teacher 1/2 to teacher 1 not to invalidate the caches when we want to train the second part of the teacher ensemble after an experiment with only one teacher. See #544

Comment (eu9ene) [reply tracking/translations_parser/utils.py:59]: So maybe it's worth adding support for the names without `/2` right away. @gregtatum We'll need to sync the work on #544 with this.

**Review by eu9ene — APPROVED** [review]
Approving but let's discuss if we want to implement support of renaming from #544 right away.

File: tracking/translations_parser/utils.py (hunk 1 above)
Comment (vrigal) [reply tracking/translations_parser/utils.py:59]: Hopefully this is already handled by the parser from #449 (covered [here](https://github.com/mozilla/firefox-translations-training/pull/589/files#diff-cc6b34cc2db0bc83dd40f5ec44f20ef4d2d81b9d01c8f919a845b00b78151b2bL22)).

|full name | corresponding run |
|-|-|
| `teacher_flores_devtest` | `teacher-1` |
| `teacher-1_flores_devtest`| `teacher-1`|
| `teacher-2_flores_devtest`| `teacher-1`|
| `teacher-flores_devtest_1/2`| `teacher-1`|
| `teacher-flores_devtest_2/2`| `teacher-2`|

**Comment by vrigal** [comment]
Closing in favor of upstream patch.

**Review by eu9ene — COMMENTED** [review]
@vrigal Do I understand correctly that now a suffix `-1` is added to all the runs? If yes, we do not want that. We train multiple models only for the teacher. This `-1` makes names longer. To be more explicit, we want:

```
(for older models)
teacher-base-0
teacher-base-1
teacher-parallel-0
teacher-parallel-1
teacher-all-0
teacher-all-1
finetune-teacher-1
finetune-teacher-2
etc.

(for newer Taskcluster models)
teacher-1
teacher-2
backwards
student
student-finetuned
quantized
teacher-ensemble
```

Naming of the teacher models will differ for the older experiments but everything else should be exactly the same across all the experiments and publishers. We might want to store a map of renamings.

**Comment by vrigal** [comment]
@eu9ene I pushed a [commit](https://github.com/mozilla/firefox-translations-training/pull/589/commits/469781d0533a713f9b794006002e81e329c1ec9d) trying to match your example.

Custom renaming could be a follow-up of this I suppose.

**Review by eu9ene — APPROVED** [review]
Ok, I see the values in unit tests but to verify this we would need to reupload everything from scratch. We can do that in #574

**Pull request merged**